File: 1.two-sum/solution-Python3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i in range(len(nums)):
            d[nums[i]] = i
        for i in range(len(nums)):
            expect_num = target - nums[i]
            if expect_num in d and i != d[expect_num]:
                return [i, d[expect_num]]

File: 1.two-sum/solution-Python3.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = HashMap<Int, Int>()
        for ((i, num) in nums.withIndex()) {
            val j = map[target - num]
            if (j != null && i != j) return intArrayOf(i, j)
            map[num] = i
        }
        return intArrayOf()
    }
}
